collection/zset/zset_test.go

```diff
@@ -493,6 +494,33 @@ func TestFloat64SetRemoveRangeByRank(t *testing.T) {
 	assert.Equal(t, N, z.Len()+len(actualNs))
 }
 
+func TestFloat64SetRemoveRangeByRankConcurrently(t *testing.T) {
+	const N = 10000
+	z := NewFloat64()
+	for i := 0; i < N; i++ {
+		z.Add(fastrand.Float64(), fmt.Sprint(i))
+	}
+
+	start, stop := func(a, b int) (int, int) {
+		if a < b {
+			return a, b
+		} else {
+			return b, a
+		}
+	}(fastrand.Intn(N), fastrand.Intn(N))
+
+	const G = 10
+	wg := sync.WaitGroup{}
+	for i := 0; i < G; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			z.RemoveRangeByRank(start, stop)
```

Comment:
How about using random start and stop values? This would allow us to remove the function(generate start and stop) above.
```suggestion
			start := fastrand.Intn(N/2)
			stop := N/2 + start
			z.RemoveRangeByRank(start, stop)
```